pkg/utils/maputils.go

```diff
@@ -65,3 +65,25 @@ func CreateRecord(columnNames []string, record []string) (map[string]interface{}
 	}
 	return recordMap, nil
 }
+
+// SetDifference returns the added and removed elements between two sets.
+func SetDifference[T1, T2 any](newSet map[string]T2, oldSet map[string]T1) ([]string, []string, bool) {
```

Comment:
add UTs for this function

Reply:
addressed in the latest commit.

Reply:
nit: this could be a little more generic like this:
```suggestion
func SetDifference[K comparable, T1, T2 any](newSet map[K]T2, oldSet map[K]T1) ([]K, []K, bool) {
```
but it's probably fine as it is since we can always make it more generic later if we actually need it to be

Reply:
addressed in the latest commit.